app/views/pulitzer/content_elements/_form.html.erb

```diff
@@ -3,19 +3,7 @@
     swf_path: image_url('uploadify.swf') } do |f| %>
   <%= f.label content_element.label %>
   <%= f.text_field :title, placeholder: 'Title' %>
-  <% if content_element.image_type? %>
-    <br>
-    <%= image_tag content_element.image_url(:thumb) %>
-    <div class="upload-group">
-      <%= f.file_field :image, placeholder: 'Image' %>
-    </div>
-  <% elsif content_element.video_type? %>
-    <%= f.text_field :body, placeholder: 'Link' %>
-    <br>
-    <%= render_video(content_element) %>
-  <% else %>
-    <%= f.text_area :body, placeholder: 'Body' %>
-  <% end %>
+  <%= content_element_input(content_element, f) %>
```

Comment:
You would also be able to remove all the helpers. By doing this you'd remove both the javascript and helper logic, and replace them with plain view files. I think it's a win for maintainability.